sencha-workspace/SlateTasksTeacher/app/view/Dashboard.js

```diff
@@ -23,6 +28,10 @@ Ext.define('SlateTasksTeacher.view.Dashboard', {
         return Ext.factory(gridLegend, 'SlateTasksTeacher.view.GridLegend', oldGridLegend);
     },
 
+    updateCourseSection: function(courseSection) {
+        this.fireEvent('coursesectionselect', courseSection);
```

Comment:
@nbey the first parameter to any component event should be a reference to the component firing the event (`this` in this case)